pkg/integrations/cadvisor/cadvisor.go

```diff
@@ -91,6 +91,10 @@ type Integration struct {
 func (i *Integration) Run(ctx context.Context) error {
 	// Do gross global configs. This works, so long as there is only one instance of the cAdvisor integration
 	// per host.
+
+	// klog
+	klog.SetLogger(i.c.logger)
```

Comment:
When running in tests, with parallelization, and the `-race` flag, this triggers a race condition that I can't seem to easily workaround. I'm tempted to remove this completely.

Reply:
Possible use of `sync.Once`?

Reply:
Yeah, that *might* have helped. Ultimately, it seems to be that cadvisor isn't properly shutting everything down, and waiting for background go routines to finish before returning. So there's still something which writes to the log after my test is "finished", and that races the next test executing and setting the logger.

That said, with some collab w/ @rfratto, there's only a single test, so we're dodging the bullet for now.